internal/utils/time_interval.go

```diff
@@ -0,0 +1,130 @@
+package utils
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const (
+	// ErrEndBeforeStart is an the error message for when a TimeInterval's end time
```

Comment:
Superfluous `the`? 

~Also, I'm from the camp which avoids over-commenting, which in this case, IMO, this would be. Sometimes unnecessary comments can be just as bad as missing comments. I do understand its a personal thing mostly.~

Scratch that, its exported. I'm blind.